### PR TITLE
Speedup travis

### PR DESCRIPTION
**Pull request by wardpeet**
Smoke tests:
- Create separate configs to reduce gatherers and audits

Travis tasks:
- Make everything concurrent

### CODE REVIEW

**Comment by wardpeet** [comment]
@paulirish sorry for taking so long...

File: .travis.yml [hunk 1]
```diff
@@ -32,14 +32,10 @@ before_script:
   - sleep 3 # wait for xvfb to boot
   - yarn build-all
 script:
-  - yarn test-cli-formatting
-  - yarn test-launcher-formatting
-  - yarn lint
-  - yarn unit
-  - yarn closure
+  - yarn test-cli-formatting & yarn test-launcher-formatting & yarn lint & yarn unit & yarn closure & yarn compile-devtools & wait
```
Comment (paulirish) [inline .travis.yml]: will a non-zero exit code in these background jobs trigger a failure?

By default I don't think it does, which is partially why i avoided doing it like so.

File: .travis.yml [hunk 2]
```diff
@@ -32,14 +32,10 @@ before_script:
...
   - yarn smoke
   - yarn smokehouse
-  - yarn compile-devtools
```
Comment (paulirish) [inline .travis.yml]: why remove this one?

Comment (wardpeet) [inline .travis.yml]: it's not removed it's also running together in the background jobs trigger

**Comment by stramel** [comment]
@wardpeet Assuming build matrix didn't work? :/

File: lighthouse-cli/test/smokehouse/offline-config.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  extends: 'lighthouse:default',
+  settings: {
+    onlyAudits: [
+      'is-on-https',
+      'redirects-http',
+      'service-worker',
+      'works-offline',
+      'viewport',
+      'without-javascript',
+      'user-timings',
```
Comment (patrickhulce) [inline lighthouse-cli/test/smokehouse/offline-config.js]: this is kind of a lot that we assert for something named "offline-config" 😆  is there anything that seems redundant we can punt?

**Comment by patrickhulce** [comment]
this is great thanks @wardpeet!

to @stramel 's build matrix point, it would be awesome to split out the smoke tests at least from the rest so we can 
* at a glance tell if it was a likely flake or real failure
* just rebuild those jobs instead of the whole shebang

File: .travis.yml (hunk 1 above)
Comment (patrickhulce) [inline .travis.yml]: how much of the time difference is a result of this change does it make reading the output a lot more difficult?

**Comment by stramel** [comment]
Is matrix not autogenerating properly with `node`, and `env`? Also, it may be useful to set `fast_finish: true` and then `allow_failures` section for flaky test suites

**Comment by wardpeet** [comment]
@stramel thanks forgot about build matrix. Yes, just experimenting for now.
It looks like https://docs.travis-ci.com/user/build-stages/ is more something I'm looking for

@brendankenny probably a good idea to do a full smoketest as well. Offline might indeed be the best one to keep a full config on.

**Comment by stramel** [comment]
@wardpeet Oh, nice! I think a combination of build matrix with build stages is exactly what we need! fail_fast should help. Allow_failures would be cool to try out for those flaky test suites

**Comment by brendankenny** [comment]
@wardpeet how are you feeling about this? Hard to follow the commits :)

If any individual optimization is causing trouble, we could do this in pieces instead, landing what's working and pushing the rest to the next PR

**Comment by stramel** [comment]
I agree with @brendankenney, I didn't mean to cause this much headache lol

**Comment by wardpeet** [comment]
@brendankenny might be better indeed, most of it is just exploring travis-ci XD. It boots up a new docker image which comes with a cost as well so looking for the best option here but I can definitely make a seperate PR for the smoke tests

**Comment by paulirish** [comment]
* Run master on node 6,7,8
* Run PRs on just node 6. 3 concurrent.
* Can adopt cloud storage for cache. Get shared across 3 concurrent workers
  * Will can help with image cleanup
* A bunch of other things i missed. :D

**Comment by wardpeet** [comment]
i'm going to close this one and create separate PRs

**Comment by paulirish** [comment]
👍  yeoman's work right here! thanks ward!
